### PR TITLE
expose `credentialsMatchesRequest(credentialItems, request)` to help …

### PR DESCRIPTION
**Pull request by jpsantosbh**
Adds function to check if a scopeRequest RESPONSE fulfill the credentials constraints on the request 

```javascript
/**
   *
   * @param credentialItems - A list of credentialItems to check
   * @param request - Original ScopeRequest
   * @return {boolean}
   */
  static credentialsMatchesRequest(credentialItems, request)
  
  //Example
  ScopeRequest.credentialsMatchesRequest(credentialItems, dsr);

### CODE REVIEW

File: src/ScopeRequest.js [hunk 1]
```diff
@@ -35,6 +35,35 @@ const isLocal = url => (url.match('(http://|https://)?(localhost|127.0.0.*)') !=
...
+   * @return {boolean}
+   */
+  static credentialsMatchesRequest(credentialItems, request) {
+    let result = true;
+    const requestedItems = _.get(request, 'credentialItems');
+    _.forEach(requestedItems, (requestedItem) => {
+      const credentialItem = _.find(credentialItems, { identifier: requestedItem.identifier });
+      if (!credentialItem) {
+        // no need to continue breaking and returning false
+        result = false;
```
Comment (lucmir) [inline src/ScopeRequest.js]: I think this line is not required.

File: src/ScopeRequest.js [hunk 2]
```diff
@@ -35,6 +35,35 @@ const isLocal = url => (url.match('(http://|https://)?(localhost|127.0.0.*)') !=
...
+        return false;
+      }
+      const verifiableCredential = VC.fromJSON(credentialItem);
+
+      const constraints = _.get(requestedItem, 'constraints');
+      const match = verifiableCredential.isMatch(constraints);
+      if (!match) {
+        // no need to continue breaking and returning false
+        result = false;
```
Comment (lucmir) [inline src/ScopeRequest.js]: I think this line can also be removed.

File: test/unit/ScopeRequest.test.js [hunk 3]
```diff
@@ -646,6 +649,44 @@ describe('DSR Request Utils', () => {
     expect(dsr).toBeDefined();
     done();
   });
+
+  it('Should ckeck is credentials matches the request constraints', () => {
```
Comment (lucmir) [inline test/unit/ScopeRequest.test.js]: Simple typo: ckeck -> check.

File: src/ScopeRequest.js (hunk 2 above)
Comment (jpsantosbh) [reply src/ScopeRequest.js]: No it's required

File: src/ScopeRequest.js (hunk 1 above)
Comment (jpsantosbh) [reply src/ScopeRequest.js]: No it's required

**Pull request merged**